diff --git a/uixml/serverlist.lua b/uixml/serverlist.lua
--- a/uixml/serverlist.lua
+++ b/uixml/serverlist.lua
@@ -76,6 +76,7 @@
 			this.connecting.Close();
 			this.connecting = nil;
 		}
+		if (Game.GatewayEnabled()) this.Widget.Visible = false;
 		OpenModal(new login(this));
 	}
 
diff --git a/uixml/login.lua b/uixml/login.lua
--- a/uixml/login.lua
+++ b/uixml/login.lua
@@ -17,6 +17,12 @@
 			}
 		});
-        scn.close.OnClick(() => this.Close());
-		this.Widget.OnEscape(() => this.Close());
+        scn.close.OnClick(() => {
+            if (Game.GatewayEnabled()) Game.Exit();
+            else this.Close();
+        });
+		this.Widget.OnEscape(() => {
+            if (Game.GatewayEnabled()) Game.Exit();
+            else this.Close();
+        });
     }
 }
