diff --git a/uixml/characterlist.lua b/uixml/characterlist.lua
index 4f45e28..a8ce0c7 100644
--- a/uixml/characterlist.lua
+++ b/uixml/characterlist.lua
@@ -4,6 +4,11 @@ class characterlist : characterlist_Designer
     {
         base()
         var e = this.Elements;
+        if (Game.GatewayEnabled()) {
+            e.mainmenu.Strid = 0;
+            e.mainmenu.InfoId = 0;
+            e.mainmenu.Text = "QUIT";
+        }
         e.listtable.SetData(Game.CharacterList());
         e.listtable.OnDoubleClick(() => Game.LoadCharacter());
         e.newchar.OnClick(() => Game.RequestNewCharacter());
@@ -12,11 +17,13 @@ class characterlist : characterlist_Designer
 		this.Widget.OnEscape(() => this.GoBack());
         e.serverlist.OnClick(() => this.ExitAnimation(() => {
             Game.StopNetworking();
-            OpenScene("serverlist");
+            if (Game.GatewayEnabled()) Game.Exit();
+            else OpenScene("serverlist");
         }));
         e.mainmenu.OnClick(() => this.ExitAnimation(() => {
             Game.StopNetworking();
-            OpenScene("mainmenu");
+            if (Game.GatewayEnabled()) Game.Exit();
+            else OpenScene("mainmenu");
         }));
     }
     
@@ -27,7 +34,8 @@ class characterlist : characterlist_Designer
 	{
 		this.ExitAnimation(() => {
             Game.StopNetworking();
-            OpenScene("serverlist");
+            if (Game.GatewayEnabled()) Game.Exit();
+            else OpenScene("serverlist");
         });
 	}
     
@@ -62,5 +70,3 @@ class characterlist : characterlist_Designer
     }
 }
 
-
-
