--- a/uixml/characterlist.lua
+++ b/uixml/characterlist.lua
@@ -5,12 +5,8 @@ class characterlist : characterlist_Designer
         e.newchar.OnClick(() => Game.RequestNewCharacter());
         e.loadchar.OnClick(() => Game.LoadCharacter());
         e.deletechar.OnClick(() => Game.DeleteCharacter());
+        e.serverlist.Visible = false;
 		this.Widget.OnEscape(() => this.GoBack());
-        e.serverlist.OnClick(() => this.ExitAnimation(() => {
-            Game.StopNetworking();
-            if (Game.GatewayEnabled()) Game.Exit();
-            else OpenScene("serverlist");
-        }));
         e.mainmenu.OnClick(() => this.ExitAnimation(() => {
             Game.StopNetworking();
             if (Game.GatewayEnabled()) Game.Exit();
