diff --git a/src/LLServer.Core/ServerApp.cs b/src/LLServer.Core/ServerApp.cs
--- a/src/LLServer.Core/ServerApp.cs
+++ b/src/LLServer.Core/ServerApp.cs
@@ -121,6 +121,6 @@
                 Path.GetFullPath(Config.NpcTransferServerCertificate, GetBasePath()), certificatePassword,
                 X509KeyStorageFlags.EphemeralKeySet);
-            npcTransferClientCaCertificate = X509Certificate2.CreateFromPemFile(
-                Path.GetFullPath(Config.NpcTransferClientCaCertificate, GetBasePath()));
+            npcTransferClientCaCertificate = X509Certificate2.CreateFromPem(
+                File.ReadAllText(Path.GetFullPath(Config.NpcTransferClientCaCertificate, GetBasePath())));
             Server.NpcTransferCertificate = npcTransferServerCertificate;
             Server.NpcTransferCaCertificate = npcTransferClientCaCertificate;
