--- a/src/LibreLancer/Server/ServerWorld.cs
+++ b/src/LibreLancer/Server/ServerWorld.cs
@@ -169,7 +169,7 @@
                     if (obj.TryGetComponent<SPlayerComponent>(out var playerComponent))
                     {
                         actions.Enqueue(() =>
-                            Server.LocalPlayer?.MissionRuntime?.LootAcquired(pickup.Nickname!, "Player"));
+                            playerComponent.Player.MissionRuntime?.LootAcquired(pickup.Nickname!, "Player"));
                     }
                 }
                 else
@@ -289,24 +289,23 @@
 
         public void LaunchComplete(GameObject obj)
         {
-            if (!string.IsNullOrWhiteSpace(obj.Nickname))
-            {
-                Server.LocalPlayer?.MissionRuntime?.LaunchComplete(obj.Nickname);
-            }
-        }
-
-        public JumperNpc[] GatherJumpers()
-        {
-            var msn = Server.LocalPlayer?.MissionRuntime;
-
-            if (msn == null)
+            if (string.IsNullOrWhiteSpace(obj.Nickname)) return;
+            var missionRuntime = obj.TryGetComponent<SPlayerComponent>(out var player)
+                ? player.Player.MissionRuntime
+                : obj.TryGetComponent<SNPCComponent>(out var npc) ? npc.MissionRuntime : null;
+            missionRuntime?.LaunchComplete(obj.Nickname);
+        }
+
+        public JumperNpc[] GatherJumpers(MissionRuntime? missionRuntime)
+        {
+            if (missionRuntime == null)
             {
                 return [];
             }
 
             var jumpers = new List<JumperNpc>();
 
-            foreach (var npc in msn.Script.Ships.Values)
+            foreach (var npc in missionRuntime.Script.Ships.Values)
             {
                 if (!npc.Jumper)
                 {
@@ -597,11 +596,11 @@
             return obj;
         }
 
-        public void SpawnJumpers(string target, JumperNpc[] jumpers)
+        public void SpawnJumpers(string target, JumperNpc[] jumpers, MissionRuntime missionRuntime)
         {
             foreach (var j in jumpers)
             {
-                NPCs.SpawnJumper(j, Server.LocalPlayer?.MissionRuntime!, target);
+                NPCs.SpawnJumper(j, missionRuntime, target);
             }
         }
 
--- a/src/LibreLancer/Server/Player.cs
+++ b/src/LibreLancer/Server/Player.cs
@@ -2214,7 +2214,7 @@
                         jumpPending = false;
                     }
                 });
-                world.DelayAction(() => { world.SpawnJumpers(target, jumpers); }, 4);
+                world.DelayAction(() => { world.SpawnJumpers(target, jumpers, MissionRuntime!); }, 4);
             }, msnPreload);
         }
 
--- a/src/LibreLancer/Server/Components/SDockableComponent.cs
+++ b/src/LibreLancer/Server/Components/SDockableComponent.cs
@@ -870,7 +870,8 @@
                 {
                     if (dock.Ship.TryGetComponent<SPlayerComponent>(out var player))
                     {
-                        player.Player.JumpTo(Action.Target!, Action.Exit!, world.Server!.GatherJumpers());
+                        player.Player.JumpTo(Action.Target!, Action.Exit!,
+                            world.Server!.GatherJumpers(player.Player.MissionRuntime));
                     }
                     else if (dock.Ship.TryGetComponent<SNPCComponent>(out var npc))
                     {
