--- a/src/LibreLancer/Server/GameServer.cs
+++ b/src/LibreLancer/Server/GameServer.cs
@@ -1767,10 +1767,11 @@
             var connectedPlayers = GetConnectedPlayers();
             foreach (var player in connectedPlayers)
             {
-                if (!(player.Space?.World?.Paused ?? false))
+                if (!player.TransferInProgress && !(player.Space?.World?.Paused ?? false))
                     player.UpdateMissionRuntime(time.TotalSeconds);
             }
             if (LocalPlayer is { } localPlayer && !connectedPlayers.Contains(localPlayer) &&
+                !localPlayer.TransferInProgress &&
                 !(localPlayer.Space?.World?.Paused ?? false))
                 localPlayer.UpdateMissionRuntime(time.TotalSeconds);
             LocalPlayer?.RunSave();
