diff --git a/src/Editor/LancerEdit/ParticleEditor/AleNodeEditor.cs b/src/Editor/LancerEdit/ParticleEditor/AleNodeEditor.cs
index c425fa11..ae76c027 100644
--- a/src/Editor/LancerEdit/ParticleEditor/AleNodeEditor.cs
+++ b/src/Editor/LancerEdit/ParticleEditor/AleNodeEditor.cs
@@ -84,7 +84,7 @@ static class AleNodeEditor
             if (ImGui.Selectable("Reverse", v == LoopFlags.Reverse) && v != LoopFlags.Reverse)
                 undo.Set("Loop", loop, LoopFlags.Reverse);
             if (ImGui.Selectable("Continue", v == LoopFlags.Continue) && v != LoopFlags.Continue)
-                undo.Set("Loop", loop, LoopFlags.PlayOnce);
+                undo.Set("Loop", loop, LoopFlags.Continue);
             ImGui.EndCombo();
         }
 
