diff --git a/src/LibreLancer/World/ProjectileManager.cs b/src/LibreLancer/World/ProjectileManager.cs
--- a/src/LibreLancer/World/ProjectileManager.cs
+++ b/src/LibreLancer/World/ProjectileManager.cs
@@ -8,6 +8,7 @@
 using System.Numerics;
 using LibreLancer.Client.Components;
 using LibreLancer.Data;
+using LibreLancer.Data.GameData;
 using LibreLancer.Data.GameData.Items;
 using LibreLancer.Fx;
 using LibreLancer.Media;
@@ -66,7 +67,7 @@
                 {
                     Projectiles[i].Alive = false;
                     Projectiles[i].Effect = null;
-                    world.Renderer?.SpawnTempFx(Projectiles[i].Data.HitEffect, contactPoint);
+                    world.SpawnTempFx(Projectiles[i].Data.HitEffect, contactPoint);
 
                     if (po?.Tag is GameObject go)
                     {
@@ -124,8 +125,7 @@
 
                 if (gunDef.Munition.Def.MunitionHitEffect != null)
                 {
-                    pdata.HitEffect = res.Items.Effects.Get(gunDef.Munition.Def.MunitionHitEffect)?
-                        .GetEffect(world.Renderer.ResourceManager);
+                    pdata.HitEffect = res.Items.Effects.Get(gunDef.Munition.Def.MunitionHitEffect);
                 }
 
                 if (gunDef.Munition.Def.ConstEffect != null)
@@ -328,7 +328,7 @@
         public required MunitionEquip Munition;
         public float Velocity;
         public float Lifetime;
-        public ParticleEffect? HitEffect;
+        public ResolvedFx? HitEffect;
         public ParticleEffect? TravelEffect;
     }
 
