--- a/src/LibreLancer/Client/CGameSession.Space.cs
+++ b/src/LibreLancer/Client/CGameSession.Space.cs
@@ -1347,24 +1347,21 @@
     {
         ModelResource? src;
         List<SeparablePart>? sep;
-        float[]? lodranges;
 
         if ((obj.Flags & ObjectSpawnFlags.Solar) == ObjectSpawnFlags.Solar)
         {
             var solar = Game.GameData.Items.Archetypes.Get(obj.Loadout.ArchetypeCrc);
             sep = solar?.SeparableParts;
             src = solar?.ModelFile?.LoadFile(Game.ResourceManager);
-            lodranges = solar?.LODRanges;
         }
         else
         {
             var ship = Game.GameData.Items.Ships.Get(obj.Loadout.ArchetypeCrc);
             sep = ship?.SeparableParts;
             src = ship?.ModelFile?.LoadFile(Game.ResourceManager);
-            lodranges = ship?.LODRanges;
         }
 
-        if (src is null || sep is null || lodranges is null)
+        if (src is null || sep is null)
             return null;
 
 #pragma warning disable CS8670
@@ -1385,9 +1382,6 @@
 
 #pragma warning restore CS8670
 
-        if (go.RenderComponent is ModelRenderer mr)
-            mr.LODRanges = lodranges;
-
         // Child damage cap
         if (sepInfo is { ChildDamageCap: not null } &&
             go.Model.TryGetHardpoint(sepInfo.ChildDamageCapHardpoint, out var capHp))
