diff --git a/src/LibreLancer/Render/DfmSkeletonManager.cs b/src/LibreLancer/Render/DfmSkeletonManager.cs
index 9d77094..78d1524 100644
--- a/src/LibreLancer/Render/DfmSkeletonManager.cs
+++ b/src/LibreLancer/Render/DfmSkeletonManager.cs
@@ -730,6 +730,38 @@ namespace LibreLancer.Render
             }
         }
 
+        public bool TryGetBoneTransform(string name, Transform3D worldTransform, out Transform3D result)
+        {
+            if (BodySkinning.Bones.TryGetValue(name, out var bodyBone))
+            {
+                result = bodyBone.LocalTransform * worldTransform;
+                return true;
+            }
+
+            if (HeadSkinning != null && HeadConnection != null && HeadSkinning.Bones.TryGetValue(name, out var headBone))
+            {
+                result = headBone.LocalTransform * HeadConnection.Transform * worldTransform;
+                return true;
+            }
+
+            if (LeftHandSkinning != null && LeftHandConnection != null &&
+                LeftHandSkinning.Bones.TryGetValue(name, out var leftHandBone))
+            {
+                result = leftHandBone.LocalTransform * LeftHandConnection.Transform * worldTransform;
+                return true;
+            }
+
+            if (RightHandSkinning != null && RightHandConnection != null &&
+                RightHandSkinning.Bones.TryGetValue(name, out var rightHandBone))
+            {
+                result = rightHandBone.LocalTransform * RightHandConnection.Transform * worldTransform;
+                return true;
+            }
+
+            result = Transform3D.Identity;
+            return false;
+        }
+
         public void StartScript(Script anmScript, float start_time, float time_scale, float duration, bool loop = false,
             float blendIn = 5f, float blendOut = 5f)
         {
