--- a/src/LibreLancer.Tests/ThnScriptDurationTests.cs
+++ b/src/LibreLancer.Tests/ThnScriptDurationTests.cs
@@ -109,6 +109,9 @@ public class ThnScriptDurationTests
 
         Assert.Empty(child.Attachments);
         Assert.Equal(new Vector3(10, 0, 0), child.GetTransform().Position);
+
+        path.Translate = new Vector3(100, 0, 0);
+        Assert.Equal(new Vector3(10, 0, 0), child.GetTransform().Position);
     }
 
     [Fact]
@@ -143,6 +146,9 @@ public class ThnScriptDurationTests
 
         Assert.Empty(child.Attachments);
         Assert.Equal(new Vector3(10, 0, 0), child.GetTransform().Position);
+
+        parent.Translate = new Vector3(100, 0, 0);
+        Assert.Equal(new Vector3(10, 0, 0), child.GetTransform().Position);
     }
 
     private sealed class ProbeEvent : ThnEvent
