diff --git a/src/LibreLancer.Data/GameItemDb.cs b/src/LibreLancer.Data/GameItemDb.cs
index c6b895f9..a9365640 100644
--- a/src/LibreLancer.Data/GameItemDb.cs
+++ b/src/LibreLancer.Data/GameItemDb.cs
@@ -116,7 +116,11 @@ public class GameItemDb
         VFS = vfs;
         var flini = new FreelancerIni(VFS);
         flData = new FreelancerData(flini, VFS);
-        ThornReadCallback = (file) => VFS.ReadAllBytes("EXE/" + file);
+        ThornReadCallback = (file) =>
+        {
+            var legacyPath = "EXE/" + file;
+            return VFS.ReadAllBytes(VFS.FileExists(legacyPath) ? legacyPath : file);
+        };
     }
 
     public string? DataPath(string? input)
diff --git a/src/LibreLancer.Data/Schema/FreelancerData.cs b/src/LibreLancer.Data/Schema/FreelancerData.cs
index f4ec92d0..97392c9b 100644
--- a/src/LibreLancer.Data/Schema/FreelancerData.cs
+++ b/src/LibreLancer.Data/Schema/FreelancerData.cs
@@ -432,8 +432,11 @@ public class FreelancerData
         Run(() =>
         {
             ContentDll = new ContentDll();
-            if (VFS.FileExists("DLLS\\BIN\\content.dll"))
-                ContentDll.Load(VFS.ReadAllBytes("DLLS\\BIN\\content.dll"));
+            var contentDllPath = VFS.FileExists("content.dll")
+                ? "content.dll"
+                : "DLLS\\BIN\\content.dll";
+            if (VFS.FileExists(contentDllPath))
+                ContentDll.Load(VFS.ReadAllBytes(contentDllPath));
             if (!string.IsNullOrEmpty(Freelancer.DataVersion))
                 DataVersion = Freelancer.DataVersion;
             else
diff --git a/src/LibreLancer.Data/Schema/FreelancerIni.cs b/src/LibreLancer.Data/Schema/FreelancerIni.cs
index 7d635a7f..d6ed7032 100644
--- a/src/LibreLancer.Data/Schema/FreelancerIni.cs
+++ b/src/LibreLancer.Data/Schema/FreelancerIni.cs
@@ -103,6 +103,8 @@ public class FreelancerIni
         if (IsLibrelancer)
         {
             DacomPath = null;
+            NewPlayerPath = "newplayer.fl";
+            MpNewCharacterPath = "mpnewcharacter.fl";
         }
 
         EquipmentPaths = [];
diff --git a/src/LibreLancer/Server/GameServer.cs b/src/LibreLancer/Server/GameServer.cs
index e6d6cce5..c74f6744 100644
--- a/src/LibreLancer/Server/GameServer.cs
+++ b/src/LibreLancer/Server/GameServer.cs
@@ -90,8 +90,8 @@ namespace LibreLancer.Server
             // TODO: initial_rep = %%FACTION%%
             // does this have any effect in FL?
 
-            var src = new StringBuilder(
-                Encoding.UTF8.GetString(FlCodec.DecodeBytes(GameData.VFS.ReadAllBytes("EXE\\mpnewcharacter.fl"))));
+            var src = new StringBuilder(Encoding.UTF8.GetString(FlCodec.DecodeBytes(
+                GameData.VFS.ReadAllBytes(GameData.Items.Ini.Freelancer.MpNewCharacterPath))));
 
             src.Replace("%%NAME%%", SavePlayer.EncodeName(name));
             src.Replace("%%BASE_COSTUME%%", pilot.Body);
