diff --git a/src/BindingsGen/GLBindingsGen/GLBindingsGen/GLBindingsGen.csproj b/src/BindingsGen/GLBindingsGen/GLBindingsGen/GLBindingsGen.csproj
index 16c12072..531f4f96 100644
--- a/src/BindingsGen/GLBindingsGen/GLBindingsGen/GLBindingsGen.csproj
+++ b/src/BindingsGen/GLBindingsGen/GLBindingsGen/GLBindingsGen.csproj
@@ -3,10 +3,7 @@
     <PropertyGroup>
         <OutputType>Exe</OutputType>
         <TargetFramework>net10.0</TargetFramework>
-    </PropertyGroup>
-
-    <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
-      <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+        <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     </PropertyGroup>
 
     <ItemGroup>
